pkg/krok/providers/cache.go.go

```diff
@@ -0,0 +1,28 @@
+package providers
+
+import (
+	"time"
+
+	"github.com/krok-o/krok/pkg/models"
+)
+
+// UserCacheItem is a user which is authenticated and doesn't need to be checked if it exists or not
+// for the duration of TTL.
+type UserCacheItem struct {
+	// constructed by time.Now().Add(TTL).
+	TTL  time.Time
+	User *models.User
+}
+
+// Expired returns if a user's TTL has expired.
+func (a *UserCacheItem) Expired() bool {
+	return time.Now().After(a.TTL)
+}
+
+// UserCache is a cache for models.User's.
+type UserCache interface {
+	Add(email string, id int)
```

Comment:
I think the cache should maybe take the whole `models.User` object in the future. Still key on the email though.

Reply:
What do you think would be the benefit of that? Keep in mind that if we would have many users that would increase the memory requirement. Not by much, but it would. :)